### DISCUSSION TITLE
Usage stats - Total connections

### DISCUSSION THREAD

**Discussion by bristoljon** [enisdenjo/graphql-ws#371] [Q&A] [answered]
### Story

As a developer/sysadmin

I want to know the total number of connected clients

So that I can establish load capacity of a single server

### Acceptance criteria

- Developer is able to log out the total connections inside onConnect (or anywhere else)

**Comment by enisdenjo** [comment, marked as answer]
Quite easy, simply use the [`onConnect`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md#onconnect) and [`onClose`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md#onclose) hooks:
```ts
import { WebSocketServer } from 'ws';
import { useServer } from 'graphql-ws/lib/use/ws';

const server = new WebSocketServer({
  port: 4000,
  path: '/graphql',
});

let connectedClients = 0;
useServer(
  {
    // ...
    onConnect: () => {
      connectedClients++;
    },
    onClose: () => {
      connectedClients--;
    },
  },
  server,
);

console.log('Listening to port 4000');
```
